README.md

```diff
@@ -182,6 +182,82 @@ You can intermittently poll whether the operation has finished by using the `Pol
   }
 ```
 
+#### `Azure::Core::Context`
+
+Most Azure SDK Service Client methods accept an optional `Azure::Core::Context` parameter, which is used to enable cancellation of the operation or to
+establish an absolute deadline for the operation.
+
+This is useful when you want to assign a time limit on an operation to ensure that it completes in a "reasonable" timeframe. For instance, the 
+snippet below will cancel a blob client upload after 500 milliseconds.
+
+```cpp
+  Azure::Core::Context cancelledIn500ms = Azure::Core::Context::ApplicationContext.WithDeadline(
+          std::chrono::system_clock::now() + std::chrono::milliseconds(500));
```

Comment:
Honestly, I would have preferred a `WithTimeout` method rather than `WithDeadline`. But that ship sailed a long time ago.

And I'm a firm believer that there should only be a single way of doing things, so while I like the idea of `WithTimeout`, we already have a `WithDeadline` which isn't *that* worse an experience.